packages/millicast-sdk/src/View.js

```diff
@@ -231,6 +239,33 @@ export default class View extends BaseWebRTC {
     this.stopReemitingWebRTCPeerInstanceEvents = reemit(webRTCPeerInstance, this, Object.values(webRTCEvents))
     this.stopReemitingSignalingInstanceEvents = reemit(signalingInstance, this, [signalingEvents.broadcastEvent])
 
+    const workerBlob = new Blob([workerString])
+    const workerURL = URL.createObjectURL(workerBlob)
+
+    webRTCPeerInstance.on('track', (trackEvent) => {
+      if (trackEvent.track?.kind !== 'video') return
+      const worker = new Worker(workerURL)
+      if (supportsRTCRtpScriptTransform) {
+        // eslint-disable-next-line no-undef
+        trackEvent.receiver.transform = new RTCRtpScriptTransform(worker, { name: 'receiverTransform' })
+      } else if (supportsInsertableStreams) {
+        const { readable, writable } = trackEvent.receiver.createEncodedStreams()
+        worker.postMessage({
+          action: 'insertable-streams-receiver',
+          readable,
+          writable
+        }, [readable, writable])
+      }
+      worker.onmessage = (event) => {
+        this.emit('onMetadata', { ...event.data, track: trackEvent.track })
```

Comment:
from below - I think this is where the parsing of the 3 fields should occur and a nicely structured + formatted object should be returned back :)